endpoint/endpointComp/endpoint.c

```diff
@@ -133,9 +133,16 @@ COMPONENT_INIT {
     device->op->get_device_id(device_id);
 #ifdef ENABLE_ENDPOINT_TEST
     LE_TEST_INIT;
+    status_t ret = SC_OK;
     LE_TEST_INFO("=== ENDPOINT TEST BEGIN ===");
-    LE_TEST(SC_OK == send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address,
-                                                  next_address, private_key, device_id_ptr, iv));
+    ret = send_transaction_information(host, port, ssl_seed, value, message, message_fmt, tag, address, next_address,
+                                       private_key, device_id_ptr, iv);
+    // Output the test status into stdout
+    if (ret == SC_OK)
```

Comment:
Don't use plain `printf`. Instead, make use of logging facilities.

Reply:
If only plain `printf` is used, it is not straightforward to redirect or inject the messages.